### PR TITLE
Added adapt features for v1.8 bundle

### PR DESCRIPTION
**Pull request by AuroraLauu**
# Template
This is a template. While not all sections are necessary, depending on the size and complexity of the PR,
more information is better. 

# @ mention of reviewers
@ckcollab @dde6khkg 
...

# A brief description of the purpose of the changes contained in this PR.
...
Added adaptation of v1.8 bundle.

# Issues this PR resolves
...
enabled v18 bundle create competition and submit solution curr

# Known issues to be addressed in a separate PR
...

# A checklist for hand testing
- [ ] add checklist here

# Any relevant files for testing
[link]('#') to any relevant files (or drag and drop into github)

# Misc. comments
...

# Checklist
- [ ] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [x] Code review by reviewer
- [x] Hand tested by reviewer
- [x] Ready to merge

### CODE REVIEW

File: src/apps/competitions/unpackers/v1.py [hunk 1]
```diff
@@ -93,33 +94,46 @@ def _unpack_phases(self):
...
+    def _preprocess_phases(self, phases):
+        """If it's version 1.8, then we remove the first phase, which is the parent phase."""
+        return phases[1:] if self._is_v18(phases) else phases
+
+    def _is_v18(self, phases):
+        """
+        Determine if the current version is 1.8
+        :return:
+        """
+        for index, phase in enumerate(phases):
```
Comment (ckcollab) [inline src/apps/competitions/unpackers/v1.py:131]: more pythonic to use `any()` probably?

```python
return any(p.get('is_parallel_parent') or p.get('parent_phasenumber') for p in phases)
```

File: src/apps/competitions/unpackers/v1.py [hunk 2]
```diff
@@ -93,33 +94,46 @@ def _unpack_phases(self):
...
+                        'file_path': os.path.join(self.temp_directory, phase[file_type]),
+                        'file_type': file_type,
+                        'creator': self.creator.id,
+                    }
+            self.competition['tasks'][task_index] = new_task
 
         self._validate_phase_ordering()
         self._set_phase_statuses()
 
+    def _preprocess_phases(self, phases):
```
Comment (ckcollab) [inline src/apps/competitions/unpackers/v1.py:122]: nice! I like this as an interface to modify phases, cool pattern

**Comment by ckcollab** [comment]
we should also add a v1.8 bundle to the functional tests, checkout this code..

Competition creation:
https://github.com/codalab/competitions-v2/blob/da5f2765a8a37a7b35f3002b7d15467db6a5582f/src/tests/functional/test_competitions.py#L50

Submission creation:
https://github.com/codalab/competitions-v2/blob/da5f2765a8a37a7b35f3002b7d15467db6a5582f/src/tests/functional/test_submissions.py#L76

Where to put test files:
https://github.com/codalab/competitions-v2/tree/da5f2765a8a37a7b35f3002b7d15467db6a5582f/src/tests/functional/test_files

File: src/apps/competitions/unpackers/v1.py (hunk 2 above)
Comment (AuroraLauu) [reply src/apps/competitions/unpackers/v1.py:122]: thanks

**Comment by AuroraLauu** [comment]
> we should also add a v1.8 bundle to the functional tests, checkout this code..
> 
> Competition creation:
> https://github.com/codalab/competitions-v2/blob/da5f2765a8a37a7b35f3002b7d15467db6a5582f/src/tests/functional/test_competitions.py#L50
> 
> Submission creation:
> https://github.com/codalab/competitions-v2/blob/da5f2765a8a37a7b35f3002b7d15467db6a5582f/src/tests/functional/test_submissions.py#L76
> 
> Where to put test files:
> https://github.com/codalab/competitions-v2/tree/da5f2765a8a37a7b35f3002b7d15467db6a5582f/src/tests/functional/test_files

i already add it, trying to fix some test failure ~

**Comment by Logan-Ruf** [comment]
Are you still working on getting these tests working?

**Comment by AuroraLauu** [comment]
> Are you still working on getting these tests working?

Yes sir

**Comment by AuroraLauu** [comment]
@ckcollab  hi Eric, CI error is being fixed

**Comment by ckcollab** [comment]
Is this ready for review now, CI appears to be passing?

**Comment by AuroraLauu** [comment]
> Is this ready for review now, CI appears to be passing?

That's awesome, please review it, tyvm!!!

**Pull request merged**